src/Microsoft.NetCore.Analyzers/UnitTests/Security/SslProtocolsAnalyzerTests.cs

```diff
@@ -465,9 +466,9 @@ public void Method()
         }
 
         [Fact]
-        public void UseTls12Or192_Diagnostic()
+        public async Task UseTls12Or192_Diagnostic()
         {
-            VerifyCSharp(@"
+            await VerifyCS.VerifyAnalyzerAsync(@"
```

Comment:
This test is failing with an inconsistent order between the 2 diagnostics. If I invert the order or the 2 diagnostics it keeps failing.

Reply:
This should only occur when the two diagnostics have the same diagnostic ID, which is not the case here. Can you try specifying the span for these diagnostics instead of the location (i.e. use `WithSpan` instead of `WithLocation`) and see if the problem goes away?

Reply:
Works perfectly fine! Thank you

Reply:
Sorry! It worked for another test not for this one. The 2 issues are reported on the exact same span.

Reply:
And reordering them did not fix it, even when you specified `WithSpan`?

Reply:
No it didn't work even after reordering. The span are the same for the 2 issues `WithSpan(9, 34, 9, 52)`.

Reply:
Tagging @dotpaul @LLLXXXCCC 

@Evangelink The security team are working off a separate `2.9.x` branch, and merge frequently into master. Your PR changing the test framework used by security analyzer unit tests is going to create merge issues between master and their branch. I think you have couple of options here:
1. Revert all the changes made to security analyzer unit tests in this PR. Then make these changes in 2.9.x branch and let it merge back into master with the automated merge PRs
2. Retarget your PR to 2.9.x branch. This might mean additional work for you to revert changes to files/tests only in master.

I think I would recommend the first approach, but will let @dotpaul make a call here.

Reply:
I will wait for @dotpaul call then :)

Reply:
Hey @Evangelink, I'd prefer the former approach that @mavasani suggested (revert changes in this PR, make a new PR tests under security/ for 2.9.x).